src/oauth/oauth2.js

```diff
@@ -77,21 +80,21 @@ export default class OAuth2 {
   exchangeForToken(oauth, userData) {
     let payload = objectExtend({}, userData)
 
-    for (let key in defaultProviderConfig.responseParams) {
-      let value = defaultProviderConfig[key]
+    for (let key in this.providerConfig.responseParams) {
+      let value = this.providerConfig.responseParams[key]
 
       switch(key) {
-        case 'code':
-          payload[key] = oauth.code
-          break
+        case 'grantType':
+            payload[value] = 'authorization_code'
```

Comment:
Also, i don't like this solution with hardcoded values. Based on OAuth 2 protocol, `grant_type` doesn't always have to be `authorization_code`.

Reply:
Shall we grab it from the providerConfig? Other flows might even require totally different requests - no?

Reply:
We could, but then it must be like this: `payload[key] = value`.

BTW, what OAuth flow you're trying to implement in your application?

Reply:
We are following the RFC: https://tools.ietf.org/html/rfc6749?#section-4.1.3

This is the way we are expecting the request to look like:

```
  POST /token HTTP/1.1
     Host: server.example.com
     Authorization: Basic czZCaGRSa3F0MzpnWDFmQmF0M2JW
     Content-Type: application/x-www-form-urlencoded

     grant_type=authorization_code&code=SplxlOBeZQQYbYS6WxSbIA
     &redirect_uri=https%3A%2F%2Fclient%2Eexample%2Ecom%2Fcb

Reply:
I'll have to think about the possible solution and get back to you tomorrow.

Reply:
Thanks for your contribution so far :)

Reply:
> I'll have to think about the possible solution and get back to you tomorrow.

Thanks a lot! Let me know if there are further things which need to be discussed!